subsys/bluetooth/mesh/adv_legacy.c

```diff
@@ -188,6 +189,21 @@ static void adv_thread(void *p1, void *p2, void *p3)
 		/* Give other threads a chance to run */
 		k_yield();
 	}
+
+	/* Empty the advertising pool when advertising is disabled */
+	while(true)
+	{
+		struct net_buf *buf;
+
+		buf = bt_mesh_adv_buf_get(K_NO_WAIT);
+
+		if (!buf) {
+			break;
+		}
+
+		bt_mesh_adv_send_start(0, -ENODEV, BT_MESH_ADV(buf));
+        net_buf_unref(buf);
+	}
```

Comment:
Small improvement:
```suggestion
	while(buf = bt_mesh_adv_buf_get(K_NO_WAIT))
	{
		bt_mesh_adv_send_start(0, -ENODEV, BT_MESH_ADV(buf));
        	net_buf_unref(buf);
	}
```
And move declaration `struct net_buf *buf;` to the beginning of the function (and removing it from line 150).

Reply:
Done!